lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1794,7 +1778,60 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
               binder.op,
               "output_padding list size does not match the number of axes");
         }
+        auto inputTensorType = cast<Torch::ValueTensorType>(input.getType());
+        if (!inputTensorType || !inputTensorType.hasSizes()) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected input type having sizes");
+        }
+        ArrayRef<int64_t> inputShape = inputTensorType.getSizes();
 
+        if (autoPad == "VALID") {
+          // Zero padding.
+          padding = defaultPadding;
+        } else if (autoPad == "NOTSET") {
+          // Explicit padding; read pads with defaults.
+          if (binder.s64IntegerArrayAttr(padding, "pads", defaultPadding))
+            return failure();
+        } else { // autopad == SAME_UPPER or SAME_LOWER
+          // Auto-padding; output_shape defaults to input_shape * strides.
+          SmallVector<int64_t> defaultOutputShape;
+          for (unsigned i = 0; i < rank - 2; i++) {
+            defaultOutputShape.push_back(inputShape[2 + i] * strides[i]);
+          }
+          if (binder.s64IntegerArrayAttr(outputShape, "output_shape",
+                                         defaultOutputShape))
+            return failure();
+          SmallVector<int64_t> paddingEnd;
+          for (unsigned i = 0; i < rank - 2; i++) {
+            int64_t totalPadding =
+                strides[i] * (inputShape[2 + i] - 1) + outputPadding[i] +
+                ((kernelShape[i] - 1) * dilations[i] + 1) - outputShape[i];
+            if (totalPadding % 2) {
+              // TODO: Add support for different padding values for the
+              // beginning and ending along each spatial axis.
+              return rewriter.notifyMatchFailure(
+                  binder.op,
+                  "unsupported conversion: the combination of stride, "
+                  "input_shape, kernel_shape, dilation, output_padding and "
+                  "output_shape caused auto-padding to produce asymmetric "
+                  "padding which isn't currently supported.");
+            }
```

Comment:
It looks like we already have a match failure for asymmetric padding below, but I think this error messaging is much better for the specific auto-pad cases that are failing, so let's keep this here.

Reply:
Agreed.